python-spec/src/somacore/query/query.py

```diff
@@ -341,8 +356,15 @@ def _read(
             for _xname in all_x_arrays
         }
 
+        # TODO: do it in parallel?
```

Comment:
Perhaps yes, as it should help with load perf, but I'd get the code structure and APIs nailed down first....

Reply:
This needs resolution. There is some risk of over-parallizing, and it will use a bit more memory to do concurrently.  But it will also be quite a bit faster given the reindexing implicit in each of these reads.

My gut feeling (not benchmarked) is to do them all concurrently (e.g., submit to the shared query thread pool).

Curious what @thetorpedodog and @johnkerl think.